components/elective/ElectiveListItem.tsx

```diff
@@ -16,18 +16,24 @@ import {
  * A List Item representing an Elective Subject.
  *
  * @param electiveSubject The Elective Subject to display.
- * @param selected Whether the Elective Subject is selected.
+ * @param radioSelected Whether the Elective Subject’s Radio is selected.
+ * @param detailSelected Whether the details of this Elective Subject are being shown.
+ * @param enrolled Whether the Student is enrolled in the Elective Subject.
  * @param onRadioToggle Triggers when the radio button is toggled.
  * @param onClick Triggers when the item is clicked.
  */
 const ElectiveListItem: StylableFC<{
   electiveSubject: ElectiveSubject;
-  selected?: boolean;
+  radioSelected?: boolean;
+  detailSelected?: boolean;
+  enrolled?: boolean;
```

Comment:
Having these list items as a state in the element would surely make it handy to get all the info in the future if needed to develop more features and see UI bugs if present.